overlord/assertstate/assertstate.go

```diff
@@ -449,14 +449,26 @@ func RefreshValidationSetAssertions(s *state.State, userID int, opts *RefreshAss
 				return fmt.Errorf("internal error: cannot check validation sets conflicts: %v", err)
 			}
 		}
-		return vsets.Conflict()
+		if err := vsets.Conflict(); err != nil {
+			return err
+		}
+
+		snaps, err := InstalledSnaps(s)
+		if err != nil {
+			return err
+		}
+		return vsets.CheckInstalledSnaps(snaps)
```

Comment:
this checks too much because it checks also the revisions which doesn't make sense, we will not have yet the right revisions, we need to ignore WrongRevisionSnaps ?